### PR TITLE
rgw: add override in header files mostly

### PR DESCRIPTION
**Pull request by Liuchang0812**
Signed-off-by: liuchang0812 <liuchang0812@gmail.com>

### CODE REVIEW

File: src/rgw/rgw_swift_auth.h [hunk 1]
```diff
@@ -152,15 +152,15 @@ class RGWHandler_SWIFT_Auth : public RGWHandler_REST {
 class RGWRESTMgr_SWIFT_Auth : public RGWRESTMgr {
 public:
   RGWRESTMgr_SWIFT_Auth() = default;
-  virtual ~RGWRESTMgr_SWIFT_Auth() = default;
+  ~RGWRESTMgr_SWIFT_Auth() override = default;
 
-  virtual RGWRESTMgr *get_resource_mgr(struct req_state* const s,
+  RGWRESTMgr *get_resource_mgr(struct req_state* const s,
```
Comment (cbodley) [inline src/rgw/rgw_swift_auth.h]: indentation

File: src/rgw/rgw_swift_auth.h [hunk 2]
```diff
@@ -152,15 +152,15 @@ class RGWHandler_SWIFT_Auth : public RGWHandler_REST {
...
                                        const std::string& uri,
                                        std::string* const out_uri) override {
     return this;
   }
 
-  virtual RGWHandler_REST* get_handler(struct req_state*,
+  RGWHandler_REST* get_handler(struct req_state*,
```
Comment (cbodley) [inline src/rgw/rgw_swift_auth.h]: indentation

File: src/rgw/rgw_rest.h [hunk 3]
```diff
@@ -391,15 +391,15 @@ class RGWRESTOp : public RGWOp {
   RGWRESTFlusher flusher;
 public:
   RGWRESTOp() : http_ret(0) {}
-  virtual void init(RGWRados *store, struct req_state *s,
-		    RGWHandler *dialect_handler) {
+  void init(RGWRados *store, struct req_state *s,
+		    RGWHandler *dialect_handler) override {
```
Comment (Liuchang0812) [inline src/rgw/rgw_rest.h]: done

File: src/rgw/rgw_rest_s3.h [hunk 4]
```diff
@@ -465,18 +465,18 @@ class RGWHandler_Auth_S3 : public RGWHandler_REST {
...
 
   static int validate_bucket_name(const string& bucket);
   static int validate_object_name(const string& bucket);
 
-  virtual int init(RGWRados *store,
+  int init(RGWRados *store,
                    struct req_state *s,
-                   rgw::io::BasicClient *cio);
-  virtual int authorize() {
+                   rgw::io::BasicClient *cio) override;
```
Comment (Liuchang0812) [inline src/rgw/rgw_rest_s3.h]: done

File: src/rgw/rgw_rest_s3.h [hunk 5]
```diff
@@ -485,28 +485,28 @@ class RGWHandler_REST_S3 : public RGWHandler_REST {
...
   RGWHandler_REST_S3() : RGWHandler_REST() {}
-  virtual ~RGWHandler_REST_S3() {}
+  ~RGWHandler_REST_S3() override {}
 
-  virtual int init(RGWRados *store,
+  int init(RGWRados *store,
                    struct req_state *s,
-                   rgw::io::BasicClient *cio);
-  virtual int authorize() {
+                   rgw::io::BasicClient *cio) override;
```
Comment (Liuchang0812) [inline src/rgw/rgw_rest_s3.h]: done

File: src/rgw/rgw_swift_auth.h (hunk 1 above)
Comment (Liuchang0812) [inline src/rgw/rgw_swift_auth.h]: done

File: src/rgw/rgw_swift_auth.h (hunk 2 above)
Comment (Liuchang0812) [inline src/rgw/rgw_swift_auth.h]: done

**Comment by Liuchang0812** [comment]
@cbodley It seems that all issues have been cleaned.

**Comment by Liuchang0812** [comment]
@cbodley hi, could we merge it now?

**Comment by cbodley** [comment]
@Liuchang0812 just trying to be careful about the timing for merge, because we'll need to rebase a lot of other PRs afterwards. what do you think, @yehudasa? is there stuff you want to get merged before this?

**Comment by Liuchang0812** [comment]
@cbodley okay

**Comment by yehudasa** [comment]
@cbodley @Liuchang0812 yeah, let's get wip-bucket-cleanup in first.

**Comment by cbodley** [comment]
@Liuchang0812 can you please rebase?

**Comment by Liuchang0812** [comment]
copy that, I will rebase it soon

2017-03-14 3:23 GMT+08:00 Casey Bodley <notifications@github.com>:

**Comment by Liuchang0812** [comment]
change log:

1. rebased to master
2. squashed all commits
3. repushed

**Pull request merged**